client/types.go

```diff
@@ -48,6 +48,26 @@ type resultOrError struct {
 	err    error
 }
 
+// Option represents an option supplied to subscription
```

Comment:
Since we actually need to find the proper type when we use them, I would take a different approach, where the options are just dumb data carriers, and we don't even use the interface defs.

```go
// Option is just used to group a limited number of concrete types, all defined in this package
type Option interface {
  isOption()
}

type OptionCapacity struct {
  Capacity int
}

func (OptionCapacity) isOption() {}
```

Usage would then be cleaner with no interface castings...

```go
	var outCapacity []int
	for _, option := range options {
	    switch option.(type) {
		case OptionCapacity:
			outCapacity = []int{option.Capacity}
	    }
	}
```

And calling simple as well

```go
c.SubscribeHeaders(ctx, headers, OptionCapacity{2})
```

Reply:
yeah. I had this idea too, not a big fan of using instance fields directly, but it's actually cleaner